lib/src/boilerplate_suggestors/advanced_props_and_state_class_migrator.dart

```diff
@@ -28,11 +29,51 @@ import 'boilerplate_utilities.dart';
 class AdvancedPropsAndStateClassMigrator extends GeneralizingAstVisitor
     with AstVisitingSuggestorMixin
     implements Suggestor {
+  final bool shouldMigrateCustomClassAndMixins;
+
+  AdvancedPropsAndStateClassMigrator(
+      {this.shouldMigrateCustomClassAndMixins = false});
+
   @override
-  visitClassDeclaration(ClassDeclaration node) {
+  void visitClassDeclaration(ClassDeclaration node) {
     super.visitClassDeclaration(node);
 
     if (!shouldMigrateAdvancedPropsAndStateClass(node)) return;
+
+    final extendsFromCustomClass = !extendsFromUiPropsOrUiState(node);
+    final hasMixins = node.withClause != null;
+
+    // Don't operate if the props class uses mixins and extends a custom class,
+    // unless the flag has been set.
+    if (hasMixins &&
+        extendsFromCustomClass &&
+        !shouldMigrateCustomClassAndMixins) return;
+
+    final className = stripPrivateGeneratedPrefix(node.name.toSource());
+    var newClassDeclarationString =
+        '\n\nclass $className = Ui${className.contains('Props') ? 'Props' : 'State'} with ';
+
+    if (extendsFromCustomClass) {
+      final parentClass = node.extendsClause.superclass.toSource() + 'Mixin';
+      newClassDeclarationString +=
+          '$parentClass, ${className}Mixin${hasMixins ? ',' : ';'}';
+    }
+
+    if (hasMixins) {
+      if (!extendsFromCustomClass) {
+        newClassDeclarationString += '${className}Mixin,';
+      }
+
+      newClassDeclarationString = node.withClause?.childEntities
+          ?.whereType<TypeName>()
+          ?.joinWithToSource(
```

Comment:
Avoid toSource. Alternatively, you could grab the name and use standard iterable utils to construct this string

```dart
final mixinNames = node.withClause.childEntities.withClause
    .mixinTypes
    .map((type) => type.name.name);
newClassDeclarationString += '${mixinNames.join(', ')};';
```


Also, using a string buffer could potentially make the code that continually appends to this string easier to read (but that's more personal preference):
```dart
newDeclarationBuffer
  ..write(mixinNames.join(', '))
  ..write(';');

```

Reply:
I added the feedback! I really like putting the logic into an extension method, so I just built the feedback into that as well, but let me know if it it's really unnecessary and I'll just move it into the suggestor 😆